rclcpp/include/rclcpp/executor.hpp

```diff
@@ -381,6 +383,47 @@ class Executor
     return FutureReturnCode::INTERRUPTED;
   }
 
+  /// Spin (blocking) until it times out, or rclcpp is interrupted.
+  /**
+   * \param[in] timeout Timeout parameter, which gets passed to Executor::spin_node_once.
+   * \throws std::runtime_error if spinning is already taking place.
+   */
+  template<typename TimeRepT = int64_t, typename TimeT = std::milli>
+  void spin_until_timeout(
```

Comment:
May I suggest rather than adding a new API, just adding a timeout field to `spin()`? Spinning without the default field will be indefinite, but if  given the timeout field, would use it and return. 

Semantically, I think that would be cleaner to use as an end-user & involve less changes to the API at the same time. 

`spin_until_XYZ_complete` I think makes sense when you're waiting for something ill-defined timeframe to complete, but when you have a specific timeout, that feels more like an argument to the main `spin` than a separate method.

Reply:
Yeah you are right, it does sound more consistant.

Reply:
I think `spin_for` is a better name for this, it matches better the standard `sleep_for()`.